### PR TITLE
Changed serviceURL and corresponding Strings to be URL types

### PR DESCRIPTION
**Pull request by ShihabMehboob**
Changed various hard coded String URLs to be URL types instead. In relation to this issue: https://github.ibm.com/arf/planning-sdk-squad/issues/499

### CODE REVIEW

File: Source/ToneAnalyzerV3/ToneAnalyzer.swift [hunk 1]
```diff
@@ -313,7 +317,7 @@ public class ToneAnalyzer {
         // construct header parameters
         var headers = defaultHeaders
         headers["Accept"] = "application/json"
-        headers["Content-Type"] = "text/plain"
+        headers["Content-Type"] = "text/html"
```
Comment (mkistler) [inline Source/ToneAnalyzerV3/ToneAnalyzer.swift]: Looks like you found a bug in the existing code -- nice!

File: Tests/LanguageTranslatorV2Tests/LanguageTranslatorTests.swift [hunk 2]
```diff
@@ -136,14 +136,14 @@ class LanguageTranslatorTests: XCTestCase {
         let creationExpectation = self.expectation(description: "Create a custom language model.")
         let deletionExpectation = self.expectation(description: "Delete the custom language model.")
 
-        #if os(iOS)
+        #if os(Linux)
+            let glossary = URL(fileURLWithPath: "Tests/LanguageTranslatorV2Tests/glossary.tmx")
+        #else
             let bundle = Bundle(for: type(of: self))
```
Comment (mkistler) [inline Tests/LanguageTranslatorV2Tests/LanguageTranslatorTests.swift]: Let's make a separate PR for the changes to the platform directives, since those don't require generator changes.

File: Source/AssistantV1/Assistant.swift [hunk 3]
```diff
@@ -23,7 +23,7 @@ import Foundation
 public class Assistant {
 
     /// The base URL to use when contacting the service.
-    public var serviceURL = "https://gateway.watsonplatform.net/assistant/api"
+    public var serviceURL = URL(string: "https://gateway.watsonplatform.net/assistant/api")
```
[the remark this answers is not in the document]
Comment (mkistler) [reply Source/AssistantV1/Assistant.swift]: Also, I think we'll need to update the section of the README.md that talks about setting the serviceURL.

File: Tests/LanguageTranslatorV2Tests/LanguageTranslatorTests.swift (hunk 2 above)
Comment (ShihabMehboob) [reply Tests/LanguageTranslatorV2Tests/LanguageTranslatorTests.swift]: This has now been raised as a new PR here: https://github.com/watson-developer-cloud/swift-sdk/pull/816

File: Source/AssistantV1/Assistant.swift [hunk 4]
```diff
@@ -195,11 +196,13 @@ public class Assistant {
             let queryParameter = URLQueryItem(name: "include_audit", value: "\(includeAudit)")
             queryParameters.append(queryParameter)
         }
-
+        
+        guard let serviceURL = serviceURL else { return }
```
[the remark this answers is not in the document]
Comment (ShihabMehboob) [reply Source/AssistantV1/Assistant.swift]: Added the following comment to all such cases: "This is unwrapping the URL, which has been determined to contain a hardcoded URL at the beginning of this file." - as it is only unwrapped as it is a URL which is optional by default as it can contain the String or nil, and the serviceURL has been populated at the beginning of the files with hardcoded values.

File: Source/AssistantV1/Assistant.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (ShihabMehboob) [reply Source/AssistantV1/Assistant.swift]: Updated the README with appropriate changes to reflect this now.

File: Source/RestKit/RestRequest.swift [hunk 5]
```diff
@@ -76,17 +76,7 @@ internal struct RestRequest {
...
         }
 
-        // construct url components
-        var urlComponents = URLComponents(string: url)!
-        if !queryItems.isEmpty {
-            urlComponents.queryItems = queryItems
-        }
-
-        // encode "+" to %2B (URLComponents does not do this)
-        urlComponents.percentEncodedQuery = urlComponents.percentEncodedQuery?.replacingOccurrences(of: "+", with: "%2B")
```
[the remark this answers is not in the document]
Comment (ShihabMehboob) [reply Source/RestKit/RestRequest.swift]: Accidentally removed this last time, thanks for spotting it. This has now been restored and amended.

**Comment by AnthonyOliveri** [comment]
@mkistler @ShihabMehboob This PR is super old now. Any objections to closing it?

**Comment by mkistler** [comment]
No objection.  Let's close.
